packages/components/src/custom-select-control-v2/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useMemo } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { useCx } from '../utils/hooks/use-cx';
+import * as Styled from './styles';
+import type {
+	CustomSelectProps,
+	CustomSelectItemProps,
+	CustomSelectContext as CustomSelectContextType,
+} from './types';
+
+export const CustomSelectContext =
+	createContext< CustomSelectContextType >( undefined );
+
+export function CustomSelect( props: CustomSelectProps ) {
+	const {
+		children,
+		defaultValue,
+		label,
+		onChange,
+		size = 'default',
+		value,
+		renderSelectedValue,
+	} = props;
+
+	const store = Ariakit.useSelectStore( {
+		setValue: ( nextValue: string ) => onChange?.( nextValue ),
+		defaultValue,
+		value,
+	} );
+
+	const { value: currentValue } = store.useState();
+
+	const cx = useCx();
+
+	const classes = useMemo(
+		() => cx( Styled.inputSize[ size ] ),
+		[ cx, size ]
+	);
+
+	return (
+		<>
+			<Ariakit.SelectLabel store={ store }>{ label }</Ariakit.SelectLabel>
+			<Styled.CustomSelectButton className={ classes } store={ store }>
+				{ renderSelectedValue
+					? renderSelectedValue( currentValue as string )
```

Comment:
I ran into the following type error for `currentValue`: 

> Argument of type 'Value' is not assignable to parameter of type 'string'.
  Type 'string[]' is not assignable to type 'string'.

Which is confusing to me since `Value` [from Ariakit is a string](https://github.com/ariakit/ariakit/blob/b26cdee02011d2a6b4b544a80842a6fa4ae9e7c5/packages/ariakit-core/src/select/select-store.ts#L29-L30)? In the meantime, I've set the `currentValue as string`

Reply:
`Value` from `ariakit` can be a `string` or a `string[]`, while `renderSelectedValue` only accept `string` as its argument. Therefore TypeScript's error is flagging that `renderSelectedValue` is not compatible with the fact that `currentValue` could also be a `string[]`.

Your current way of removing the error is valid, and there would also be other alternatives — for example, adding checks to [narrow down](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) the type of `currentValue`, making sure that it isn't an array.

At the same time, this is starting to feel like enough work to justify supporting multiple selection right from the start.

Reply:
Marking this as resolved, since multiple selection is now supported 🥳